lib/ruby_lsp/response_builders/hover.rb

```diff
@@ -12,27 +12,39 @@ class Hover < ResponseBuilder
       sig { void }
       def initialize
         super
-        @stack = T.let(
-          [],
-          T::Array[String],
+
+        @response = T.let(
+          {
+            title: +"",
+            signature: +"",
+            links: +"",
+            documentation: +"",
+          },
+          T::Hash[Symbol, String],
         )
       end
 
-      sig { params(hover_response: String).void }
-      def push(hover_response)
-        @stack << hover_response
+      sig { params(content: String, category: Symbol).void }
+      def push(content, category)
+        hover_content = @response[category]
+        if hover_content
+          hover_content << content + "\n"
+        end
       end
 
-      alias_method(:<<, :push)
-
       sig { returns(T::Boolean) }
       def empty?
-        @stack.empty?
+        @response.all? { |_key, value| value.empty? }
       end
 
-      sig { override.returns(String) }
+      sig { override.returns(ResponseType) }
       def response
-        @stack.join("\n\n")
+        result = T.must(@response[:title])
+        result << @response[:signature] if @response[:signature]
```

Comment:
When you mentioned that this would require us to remove the code blocks where we add it, are you essentially suggesting we swap all instances of `title` with `signature`, then remove the additional `signature` instances?

Reply:
I mean changing `title` to `signature`, then removing the "\`\`\`ruby...```" blocks from places like [here](https://github.com/Shopify/ruby-lsp/blob/9159861f94da242fb49383bca3f7ccaa9022df0f/lib/ruby_lsp/requests/support/common.rb#L92) and adding those blocks on this line I commented on.

That way, all content added to `signature` is automatically assumed to be Ruby code and inserted inside the Ruby block.